modules/renter/upload.go

```diff
@@ -62,47 +62,46 @@ func (r *Renter) threadedUploadPiece(up modules.FileUploadParams, piece *filePie
 		// Select a host. An error here is unrecoverable.
 		host, err := r.hostDB.RandomHost()
 		if err != nil {
-			return
+			return err
 		}
 
 		// Negotiate the contract with the host. If the negotiation is
-		// unsuccessful, we need to try again with a new host. Otherwise, the
-		// file will be uploaded and we'll be done.
+		// unsuccessful, we need to try again with a new host.
 		contract, contractID, key, err := r.negotiateContract(host, up)
-		if err != nil {
-			// The previous attempt didn't work. We will try again after
-			// sleeping for a randomized amount of time to increase our chances
-			// of success. This will help spread things out if there are
-			// problems with network congestion or other randomized issues.
-			randSource := make([]byte, 1)
-			rand.Read(randSource)
-			time.Sleep(time.Duration(attempts) * time.Duration(attempts) * 250 * time.Millisecond * time.Duration(randSource[0]))
-			continue
+		if err == nil {
```

Comment:
Normally I wouldn't have touched it but I thought I was going to modify the function more. This control flow made more sense for the modifications I had in mind.
I still think it's slightly better this way, since the sleep is associated with the loop instead of the error condition. idk, it weirds me out to have the success condition in the main body of the loop.